### PR TITLE
feat(templates): make package scripts more windows friendly, add mud:up

### LINKED ISSUE #831: add foundryup command to templates?
like this: https://github.com/latticexyz/mud/blob/843c663bb8f69e5aad5e84089fbe77e3f0f726b3/package.json#L17
(might need to double check if this works on windows first though, I suspect `curl` and `bash` doesn't)

### LINKED ISSUE #832: templates' `dev` script should `pnpm initialize` before running `run-pty`
otherwise the client will error with missing world types

### PR DESCRIPTION
**Pull request by alvrs**
closes #832
closes #831

- Windows users are reporting issues with `node-pty`. This PR adds a simple `dev` script relying on `pnpm recursive` to remove the need for `run-pty`/`node-pty`
  - I think we should split the `dev:node` command from this dev script because if we have time I'd like to add a simple file watcher that runs codegen and redeploys the contracts when relevant files change, but we don't need to restart anvil for that.
  - ~~we still have the `dev:pty` script which is recommended for everyone who doesn't have issues with `node-pty` because it makes it easier to see when things fail and why they fail~~ removing the `run-pty` dependency because it causes issues for the initial install

- this PR also adds a simple `mud:up` script to update mud to the latest canary
- this PR also adds mud cli as a workspace dependency for templates, so users can run `pnpm mud` anywhere in their project instead of just in the contracts dir

### CODE REVIEW

File: templates/vanilla/package.json [hunk 1]
```diff
@@ -3,14 +3,17 @@
...
     "dev:contracts": "pnpm --filter 'contracts' dev",
-    "dev:node": "pnpm --filter 'contracts' devnode",
+    "dev:node": "mud devnode",
+    "dev:pty": "run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
     "initialize": "pnpm recursive run initialize",
+    "mud:up": "pnpm recursive exec mud set-version -v canary && pnpm install",
     "test": "pnpm recursive run test"
   },
   "devDependencies": {
+    "@latticexyz/cli": "link:../../packages/cli",
```
Comment (alvrs) [inline templates/vanilla/package.json:16]: @holic just double checking, will this also get replaced with the correct version when installing the template?

File: templates/react/package.json [hunk 2]
```diff
@@ -3,14 +3,17 @@
   "private": true,
   "scripts": {
     "build": "pnpm recursive run build",
-    "dev": "run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
+    "dev": "pnpm recursive run dev",
```
Comment (holic) [inline templates/react/package.json:6]: while you're here, can you add `initialize` before recursive `dev`?
```suggestion
    "dev": "pnpm run initialize && pnpm recursive run dev",
```

File: templates/vanilla/package.json (hunk 1 above)
Comment (holic) [reply templates/vanilla/package.json:16]: I think so! https://github.com/latticexyz/mud/blob/main/packages/create-mud/scripts/copy-templates.sh

You can test by running `pnpm build` inside of `create-mud` and look at `dist/templates`

File: templates/vanilla/package.json [hunk 3]
```diff
@@ -3,14 +3,17 @@
   "private": true,
   "scripts": {
     "build": "pnpm recursive run build",
-    "dev": "run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
+    "dev": "pnpm recursive run dev",
```
Comment (alvrs) [inline templates/vanilla/package.json:6]:
```suggestion
    "dev": "pnpm run initialize && recursive run dev",
```

Comment (holic) [inline templates/vanilla/package.json:6]:
```suggestion
    "dev": "pnpm run initialize && pnpm recursive run dev",
```

File: templates/vanilla/package.json (hunk 1 above)
Comment (alvrs) [reply templates/vanilla/package.json:16]: yep that worked!

**Review by holic — DISMISSED** [review]

File: templates/react/package.json [hunk 4]
```diff
@@ -3,14 +3,17 @@
   "private": true,
   "scripts": {
     "build": "pnpm recursive run build",
-    "dev": "run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
+    "dev": "pnpm run initialize && pnpm recursive run dev",
     "dev:client": "pnpm --filter 'client' run dev",
     "dev:contracts": "pnpm --filter 'contracts' dev",
-    "dev:node": "pnpm --filter 'contracts' devnode",
+    "dev:node": "mud devnode",
+    "dev:pty": "run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
```
Comment (alvrs) [inline templates/react/package.json:10]: actually we probably want to add that here too
```suggestion
    "dev:pty": "pnpm run initialize && run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
```

File: templates/vanilla/package.json [hunk 5]
```diff
@@ -3,14 +3,17 @@
   "private": true,
   "scripts": {
     "build": "pnpm recursive run build",
-    "dev": "run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
+    "dev": "pnpm run initialize && recursive run dev",
     "dev:client": "pnpm --filter 'client' run dev",
     "dev:contracts": "pnpm --filter 'contracts' dev",
-    "dev:node": "pnpm --filter 'contracts' devnode",
+    "dev:node": "mud devnode",
+    "dev:pty": "run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
```
Comment (alvrs) [inline templates/vanilla/package.json:10]:
```suggestion
    "dev:pty": "pnpm run initialize && run-pty % pnpm dev:node % pnpm dev:client % pnpm dev:contracts",
```

**Review by holic — APPROVED** [review]
🎉

**Pull request merged**